pkg/controllers/internalmembercluster/v1beta1/member_controller_test.go

```diff
@@ -1178,3 +1185,296 @@ func TestUpdateResourceStats(t *testing.T) {
+
+func (p *startTimedOutProvider) Start(_ context.Context, _ *rest.Config) error {
+	<-p.hold
+	return nil
+}
+
+func (p *startTimedOutProvider) Collect(_ context.Context) propertyprovider.PropertyCollectionResponse {
+	return propertyprovider.PropertyCollectionResponse{}
+}
+
+func TestConnectToPropertyProvider(t *testing.T) {
+	imcTemplate := &clusterv1beta1.InternalMemberCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: imcName,
+		},
+		Spec: clusterv1beta1.InternalMemberClusterSpec{},
+	}
+
+	// Note that to verify the correctness of the behavior, the data here is set to be
+	// explicitly different from the those returned by the dummy property provider.
+	nodes := []*corev1.Node{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: nodeName1,
+			},
+			Spec: corev1.NodeSpec{},
+			Status: corev1.NodeStatus{
+				Capacity: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("10"),
+					corev1.ResourceMemory: resource.MustParse("10Gi"),
+				},
+				Allocatable: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("8"),
+					corev1.ResourceMemory: resource.MustParse("8Gi"),
+				},
+			},
+		},
+	}
+	pods := []*corev1.Pod{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: podName1,
+			},
+			Spec: corev1.PodSpec{
+				NodeName: nodeName1,
+				Containers: []corev1.Container{
+					{
+						Name: containerName1,
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
+								corev1.ResourceCPU:    resource.MustParse("2"),
+								corev1.ResourceMemory: resource.MustParse("1Gi"),
+							},
+						},
+					},
+					{
+						Name: containerName2,
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
+								corev1.ResourceCPU:    resource.MustParse("1"),
+								corev1.ResourceMemory: resource.MustParse("2Gi"),
+							},
+						},
+					},
+				},
+			},
+			Status: corev1.PodStatus{
+				Phase: corev1.PodRunning,
+			},
+		},
+	}
+
+	h := make(chan struct{})
+	// Always close the channel to avoid leaks.
+	defer func() {
+		close(h)
+	}()
+
+	testCases := []struct {
+		name             string
+		propertyProvider propertyprovider.PropertyProvider
+		wantIMCStatus    clusterv1beta1.InternalMemberClusterStatus
+	}{
+		{
+			name:             "provider failed to start",
+			propertyProvider: &failedToStartProvider{},
+			wantIMCStatus: clusterv1beta1.InternalMemberClusterStatus{
+				Properties: map[clusterv1beta1.PropertyName]clusterv1beta1.PropertyValue{
+					propertyprovider.NodeCountProperty: {
+						Value: "1",
+					},
+				},
+				ResourceUsage: clusterv1beta1.ResourceUsage{
+					Capacity: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("10"),
+						corev1.ResourceMemory: resource.MustParse("10Gi"),
+					},
+					Allocatable: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("8"),
+						corev1.ResourceMemory: resource.MustParse("8Gi"),
+					},
```

Comment:
Can we use a variable to link the node's capacity/allocatable with the value here? IIRC, they must match

Reply:
Done.